subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -183,6 +188,30 @@ private SnapshotHierarchy invalidateUnsupportedFileSystems(SnapshotHierarchy roo
         }
     }
 
+    @CheckReturnValue
+    private SnapshotHierarchy updateUnwatchableFilesOnBuildStart(SnapshotHierarchy root, Invalidator invalidator, WatchMode watchMode) {
+        SnapshotHierarchy newRoot = root;
+        // Make sure we don't have any unwatchable files in the VFS, since the some previously unwatchable files may become watchable.
+        newRoot = invalidateUnwatchableHierarchies(newRoot, invalidator, unwatchableFiles);
+        if (shouldWatchUnsupportedFileSystems(watchMode)) {
+            unwatchableFiles = FileHierarchySet.empty();
+            // Replay the watchable hierarchies since the end of last build, since they have become watchable.
```

Comment:
This is about the "unwatchable" ones, no?

Reply:
It's complicated ;) Those are the ignored watchable hierarchies. I suppose I should also rename the field to `ignoredWatchableHierarchiesSinceLastBuildFinish `